src/agent/driver-test-runner.js

```diff
@@ -91,27 +91,37 @@ export class DriverTestRunner {
   }
 
   /**
-   * @param {"reading" | "interactive"} mode
+   * Used for v2 aria-at tests, "reading" and "interactive" map to various settings based on AT.
+   *
+   * @param {string} settings - "browseMode" "focusMode" for NVDA, "pcCursor" "virtualCursor"
+   *                 for JAWS., "defaultMode" for others.
    */
-  async ensureMode(mode) {
-    const capabilities = await this.collectedCapabilities;
-    if (capabilities.atName == 'NVDA') {
+  async ensureSettings(settings) {
+    const { atName } = await this.collectedCapabilities;
+    if (atName == 'NVDA') {
       // disable the "beeps" when switching focus/browse mode, forces it to speak the mode after switching
       await this.atDriver._send({
         method: 'nvda:settings.setSettings',
         params: { settings: [{ name: 'virtualBuffers.passThroughAudioIndication', value: false }] },
       });
       try {
-        const desiredResponse = mode.toLowerCase() === 'reading' ? 'Browse mode' : 'Focus mode';
+        const desiredResponse = { browsemode: 'Browse mode', focusmode: 'Focus mode' }[
+          settings.toLowerCase()
+        ];
+        if (!desiredResponse) {
+          throw new Error(`Unknown command settings for NVDA "${settings}"`);
+        }
         const MODE_SWITCH_SPEECH_TIMEOUT = 500;
         let speechResponse;
+        // there are 2 modes, so we will try pressing mode switch up to twice
         for (let triesRemain = 2; triesRemain > 0; triesRemain--) {
           speechResponse = await this._collectSpeech(MODE_SWITCH_SPEECH_TIMEOUT, () =>
             this.sendKeys(ATKey.sequence(ATKey.chord(ATKey.key('insert'), ATKey.key('space'))))
           );
+          // convert from array of strings to single string and trim begin/end whitespace
           speechResponse = speechResponse.join(' ').replace(/^\s+|\s+$/g, '');
```

Comment:
For future reference, `String.prototype.trim` would obviate an explanation

Reply:
Changed to .trim() in dbebce3